### PR TITLE
ANSA-288: Refactor of skip logic

### PR DESCRIPTION
**Pull request by tmvumbi2**

### CODE REVIEW

File: src/utils/skipLogic.js [hunk 1]
```diff
@@ -123,12 +123,12 @@ module.exports = function () {
...
   // TODO do a complex (more than one depth) groups, probably xor xnor would handle this
-  var skipLogicGroupCheck = function (scope, value, check) {
+  var skipLogicGroupCheck = function (scope, check) {
     var logicGate = check.logicGate
     var checkEval = null
     for (var i = 0; i < check.group.length; i++) {
       var c = check.group[i]
-      value = scope.form && c.variable ? scope.form[c.variable].$modelValue : value
+      var value = scope.form[c.variable]['$modelValue']
       if (checkEval !== null) {
```
Comment (kaweesi) [inline src/utils/skipLogic.js]: wouldn't this be scope.form[c.variable].$modelValue

File: test/utils/skipLogic.js [hunk 2]
```diff
@@ -300,8 +300,14 @@ tap.test('.skipLogic()', { autoend: true }, (t) => {
         settings: {
           showhide: null
         }
-      }
+      },
+      form: []
     }
+
+    scope.form['var1'] = { $modelValue: 'yes' }
+    scope.form['var2'] = { $modelValue: 'no' }
```
Comment (kaweesi) [inline test/utils/skipLogic.js]: this woudn't then change to scope.form['var1'].$modelValue = 'yes' forexample

File: test/utils/skipLogic.js [hunk 3]
```diff
@@ -320,30 +326,23 @@ tap.test('.skipLogic()', { autoend: true }, (t) => {
...
-    t.ok(scope.field.show)
-    skipLogicGroupCheck(scope, 'no', check)
+    skipLogicGroupCheck(scope, check)
     t.ok(scope.field.show)
 
+    scope.form.var2.$modelValue = 'yes'
     check.logicGate = 'and'
-    check.group[1].value = 'yes'
-    skipLogicGroupCheck(scope, 'yes', check)
+    skipLogicGroupCheck(scope, check)
```
Comment (kaweesi) [inline test/utils/skipLogic.js]: i am not sure why you are deciding to change these testcases!!! they would still pass if you resolve my above reviews @tmvumbi2

File: test/utils/skipLogic.js [hunk 4]
```diff
@@ -320,30 +326,23 @@ tap.test('.skipLogic()', { autoend: true }, (t) => {
       }]
     }
 
-    skipLogicGroupCheck(scope, 'yes', check)
+    skipLogicGroupCheck(scope, check)
     t.notOk(scope.field.show)
 
     check.logicGate = 'or'
-    skipLogicGroupCheck(scope, 'yes', check)
```
Comment (kaweesi) [inline test/utils/skipLogic.js]: you would just remove the check after the var initialisations and that should still have these tests pass

**Comment by BMartinos** [comment]
Thanks Guys! From my side this all looks good, Id just suggest to move split the tests into seperate logicGate tests to clearly show the AND / OR functionality. It just a tiny bit hard to follow with joining these tests into one

File: test/utils/skipLogic.js [hunk 5]
```diff
@@ -294,30 +300,55 @@ tap.test('.skipLogic()', { autoend: true }, (t) => {
...
+    scope.form['var2'] = { $modelValue: 'no' }
+    scope.form['var3'] = { $modelValue: 'yes' }
+
+    const check = {
+      logicGate: 'or',
+      action: 'showhide',
+      group: [{
+        operand: '=',
+        value: 'yes',
+        variable: 'var1'
```
Comment (kaweesi) [inline test/utils/skipLogic.js]: probably declaring this before the form variable setters above would make this test case more useful.

File: test/utils/skipLogic.js [hunk 6]
```diff
@@ -268,14 +268,20 @@ tap.test('.skipLogic()', { autoend: true }, (t) => {
     t.end()
   })
 
-  t.test('skipLogicGroupCheck(): should match one depth group appropriately', (t) => {
+  t.test('skipLogicGroupCheck(): should match one depth group appropriately with "and" logicGate', (t) => {
     const scope = {
```
Comment (kaweesi) [inline test/utils/skipLogic.js]: in this and test, we need to test when it's not ok

File: test/utils/skipLogic.js [hunk 7]
```diff
@@ -294,30 +300,55 @@ tap.test('.skipLogic()', { autoend: true }, (t) => {
...
+        }
+      },
+      form: []
+    }
+
+    scope.form['var1'] = { $modelValue: 'yes' }
+    scope.form['var2'] = { $modelValue: 'no' }
+    scope.form['var3'] = { $modelValue: 'yes' }
+
+    const check = {
```
Comment (kaweesi) [inline test/utils/skipLogic.js]: same applies here, probably declaring these before setting form variables makes the test more understandable
